apps/src/code-studio/pd/foorm/library_editor/FoormLibrarySaveBar.jsx

```diff
@@ -259,71 +277,122 @@ class FoormSaveBar extends Component {
     );
   }
 
-  renderNewFormSaveModal = () => {
-    const showFormNameError = this.state.formName && !this.isFormNameValid();
+  showNameError = () =>
+    (this.state.libraryQuestionName &&
+      !this.isNameValid(this.state.libraryQuestionName)) ||
+    (this.state.libraryName && !this.isNameValid(this.state.libraryName));
+
+  disableSaveNewLibraryQuestion = () => {
+    // disable if invalid name for library question
+    if (this.showNameError()) {
+      return true;
+    }
+
+    // for existing library, library question name is all that is required
+    if (this.props.libraryId && !this.state.libraryQuestionName) {
+      return true;
+    }
+
+    // for new library, need library name, category, and a library question name
+    if (
+      !this.props.libraryId &&
+      !(
+        this.state.libraryName &&
+        this.state.libraryCategory &&
+        this.state.libraryQuestionName
+      )
+    ) {
+      return true;
+    }
+
+    return false;
+  };
+
+  renderNewLibraryQuestionSaveModal = () => {
```

Comment:
This `FoormLibrarySaveBar` component is getting really big, it might be worth factoring `NewLibraryQuestionSaveModal` into a separate component

Reply:
Going to leave this as a to-do in the spirit of pushing this across the finish line and breaking work into smaller PRs in the future, if that's ok with you.

Reply:
out of curiosity, what's the rush to get this across the finish line? it's my experience that if you leave this as a to-do task it may never get done.

in the spirit of breaking things into smaller PRs, you could always do it in a branch based off this branch then open a PR into this branch.

that said, might be good to think about if/how we're going to refactor to share more code between Foorm and FoormLibrary before refactoring this, so i'm totally fine with deferring.

Reply:
Re: additional changes, added a general comment about responding to everyone's feedback, LMK what you think.

Re: timeline, I don't know if there's a hard deadline -- we have a bug bash scheduled for the end of this week (could be pushed), and had set a goal of being able to edit surveys for some spring workshops via this system. They're not big changes though and could easily be done via our old approach to making changes to Foorm surveys/libraries (via PR).